compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2809,11 +2809,13 @@ object Parsers {
       if (isIdent(nme.raw.BAR)) { in.nextToken(); pattern1(location) :: patternAlts(location) }
       else Nil
 
-    /**  Pattern1     ::= Pattern2 [Ascription]
+    /**  Pattern1     ::= PatVar Ascription
+     *                  | SimpleLiteral Ascription
+     *                  | Pattern2
      */
     def pattern1(location: Location = Location.InPattern): Tree =
       val p = pattern2()
-      if in.isColon then
+      if (isVarPattern(p) || p.isInstanceOf[Number]) && in.isColon then
```

Comment:
The code does not match the syntax: simple literals can also be strings, characters, or booleans. Also there should be tests that all simple literals can be given a type ascription.

Reply:
The numeric literal tests are in your genericNumLits, BigFloat, GenericNumLits tests.